<docs>
	<doc for="InvalidOperationException.InvalidOperationException3">
		<summary>
			<para>Initializes a new instance of the <see cref="System.InvalidOperationException"/> class with serialized data.</para>
		</summary>
		<param name="info">The object that holds the serialized object data.</param>
		<param name="context">The contextual information about the source or destination.</param>
		<remarks>
			<para>This constructor is called during deserialization to reconstitute the
      exception object transmitted over a stream. For more information, see
   <see topic="cpconSerialization"/>.</para>
		</remarks>
		<seealso topic="cpconSerialization"/>
	</doc><doc for="InvalidOperationException.InvalidOperationException">
		<overload>
			<para>Initializes a new instance of the
   <see cref="System.InvalidOperationException"/> class.</para>
		</overload>
		<summary>
   Initializes a new instance of the
<see cref="System.InvalidOperationException"/> class.
</summary>
		<remarks>
			<para>This constructor initializes the <see cref="System.Exception.Message"/> 
property of the new instance to a system-supplied message that describes the
error, such as "The requested operation cannot be performed." This message takes
into account the current system culture.</para>
			<para>The following table shows the initial property values for an instance of <see cref="System.InvalidOperationException"/>.</para>
			<list type="table"><listheader>
					<term>Property</term>
					<description>Value</description>
				</listheader>
				<item>
					<term><see cref="System.Exception.InnerException"/></term>
					<description>A 
   null reference (<see langword="Nothing"/> in Visual Basic).</description>
				</item>
				<item>
					<term><see cref="System.Exception.Message"/></term>
					<description>The localized error message string.</description>
				</item>
			</list>
		</remarks>
	</doc><doc for="InvalidOperationException.InvalidOperationException2">
		<summary>
			<para>Initializes a new instance of the <see cref="System.InvalidOperationException"/>
class with a specified error message and a reference to the
inner exception that is the cause of this exception.</para>
		</summary>
		<param name="message">The error message that explains the reason for the exception.</param>
		<param name="innerException">The exception that is the cause of the current exception. If the <paramref name="innerException"/> parameter is not a null reference (<see langword="Nothing"/> in Visual Basic), the current exception is raised in a <see langword="catch"/> block that handles the inner exception.</param>
		<remarks>
			<para>An exception that is thrown as a direct result of a
      previous exception should include a reference to the previous exception in
      the <see cref="System.Exception.InnerException"/> property. The
   <see cref="System.Exception.InnerException"/> property returns 
      the same value that is passed into the constructor, or a null reference
      (<see langword="Nothing"/> in Visual Basic) if the <see cref="System.Exception.InnerException"/>
      property does not supply the inner exception value to
      the constructor. </para>
			<para>The following table shows the initial property values for an instance of
   <see cref="System.InvalidOperationException"/>.</para>
			<list type="table"><listheader>
					<term>Property</term>
					<description>Value</description>
				</listheader>
				<item>
					<term><see cref="System.Exception.InnerException"/></term>
					<description>The inner exception reference.</description>
				</item>
				<item>
					<term><see cref="System.Exception.Message"/></term>
					<description>The error message string.</description>
				</item>
			</list>
		</remarks>
		<seealso cref="System.Exception"/>
		<seealso topic="cpconHandlingThrowingExceptions"/>
	</doc><doc for="InvalidOperationException.InvalidOperationException1">
		<summary>
			<para>Initializes a new instance of the
   <see cref="System.InvalidOperationException"/> class with
      a specified error message.</para>
		</summary>
		<param name="message">The message that describes the error.</param>
		<remarks>
			<para>This constructor initializes the <see cref="System.Exception.Message" qualify="true"/> 
property of the new exception using the <paramref name="message "/>parameter. The content of <paramref name="message"/> is intended to be
understood by humans. The caller of this constructor is required to ensure that
this string has been localized for the current system culture.</para>
			<para>The following table shows the initial property values for an instance of <see cref="System.InvalidOperationException"/>.</para>
			<list type="table"><listheader>
					<term>Property</term>
					<description>Value</description>
				</listheader>
				<item>
					<term><see cref="System.Exception.InnerException"/></term>
					<description>A
   null reference (<see langword="Nothing"/> in Visual Basic).</description>
				</item>
				<item>
					<term><see cref="System.Exception.Message"/></term>
					<description>The error message string.</description>
				</item>
			</list>
		</remarks>
	</doc><doc for="InvalidOperationException">
		<summary>
			<para> The exception that is thrown when a method call is invalid for the object's current
      state.</para>
		</summary>
		<remarks>
			<para><see cref="System.InvalidOperationException"/> is used in cases when the
   failure to invoke a method is caused by reasons other than invalid arguments. For
   example, <see cref="System.InvalidOperationException"/>
   is thrown by:</para>
			<list type="bullet"><item>
					<term>
						<see cref="System.Collections.IEnumerator.MoveNext"/>
   if objects of a collection are modified after the enumerator is created.</term>
				</item>
			</list>
			<list type="bullet"><item>
					<term>
						<see cref="System.Resources.ResourceSet.GetString"/>
   if the resource set is closed before the method call is made.</term>
				</item>
			</list>
			<para>If the method invocation failure is due to invalid
   arguments, then <see cref="System.ArgumentException"/> or
   one of its derived classes, <see cref="System.ArgumentNullException"/>
   or <see cref="System.ArgumentOutOfRangeException"/>, should
   be thrown instead.</para>
			<list type="bullet"><item>
					<term>
      
      The <see cref="System.Reflection.Emit.OpCodes.Ldflda"/> Microsoft intermediate
      language (MSIL) instruction throws <see cref="System.InvalidOperationException"/>.</term>
				</item>
			</list>
			<para><see cref="System.InvalidOperationException"/> uses the HRESULT
COR_E_INVALIDOPERATION, which has the value 0x80131509. </para>
			<para>For a list of initial property values for an instance of <see cref="System.InvalidOperationException"/>, see the <see cref="System.InvalidOperationException()"/> constructors. </para>
		</remarks>
		<seealso cref="System.Exception"/>
		<seealso cref="System.Collections.IEnumerator"/>
		<seealso cref="System.Resources.ResourceSet"/>
		<seealso topic="cpconHandlingThrowingExceptions"/>
	</doc>
</docs>
